### PR TITLE
Update TNO interface definition

### PR DESCRIPTION
**Pull request by dave992**
Update the interface definition for TNO after the first integration testing at SAM XL.

Note: the changes are currently a direct copy of the TNO changes, consider this MR therefore made by TNO. 

## Motivation and Context
TNO uses additional fields and messages in their interface. 

## Changes
- Add a trigger state
- Update the `ResultCode` and `ModuleState`

## Checklist
- [x] Update `REAMDE.md` 
- [x] Update version

### CODE REVIEW

File: penelope_aerospace_pl_msgs/msg/TriggersState.msg [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#-------------------------------------------------------------------------------
+# DVM Triggers State (TNO)
+#-------------------------------------------------------------------------------
+
+# Description of the DVM process states:
+#   GIVE_TRIGGERS
+#       ...
+#   STOP_GIVING_TRIGGERS
+#       ...
```
Comment (dave992) [inline penelope_aerospace_pl_msgs/msg/TriggersState.msg:9]: I would suggest using a boolean value for this in the action definition, something like:
```
# Indicate if the DVM acquisition triggers are active 
#   True if a motion request with triggers for the DVM system is sent to the manipulator
#   and is being processed. False in all other situations.

bool triggers_active
```

If you intend to have more states, you could keep this message, but I would add the other states and update the naming as the verb seems to imply this is requesting triggers. 

I would suggest the following:
```suggestion
#   TRIGGERS_ACTIVE
#        A motion request with triggers for the DVM system is sent to the manipulator
#        and is being processed.
#   NO_TRIGGERS
#       No motion request with triggers is active or being processed. This is a valid
#       state for motions without triggers or while no motion is requested. 
```

File: penelope_aerospace_pl_msgs/msg/ResultCodes.msg [hunk 2]
```diff
@@ -10,7 +10,8 @@ string SUCCES=""
 uint16 RC_SUCCES=1
 
 # Generic failure message
-string FAILED="Action/Service failed.. See module log for more details."
+string FAILED_BY_USER="Action/Service failed scan, terminated by operator."
+string FAILED="Action/Service failed scan with errors."
 uint16 RC_FAILED=2
```
Comment (dave992) [inline penelope_aerospace_pl_msgs/msg/ResultCodes.msg:16]: The `FAILED` string and `RC_FAILED` are fallback options for others too. I would create specific result codes and messages for TNO:

```
# DVM status messages (Range between 100 and 199)
string DVM_FAILED_BY_USER="DVM scan failed, terminated by operator."
uint16 RC_DVM_FAILED_BY_USER=100

string FAILED_BY_USER="Action/Service failed scan, terminated by operator."
```

File: penelope_aerospace_pl_msgs/msg/ModuleState.msg [hunk 3]
```diff
@@ -16,5 +16,6 @@ uint8 ACCEPTED=1
 uint8 EXECUTING=2
 uint8 PAUSED=7
 uint8 CANCELING=3
+uint8 INITIALIZING=4
```
Comment (dave992) [inline penelope_aerospace_pl_msgs/msg/ModuleState.msg:19]: This should go in the `DVMState`. The `ModuleState` is only sent in feedback messages and is shared across all modules. I have not seen any module send feedback while still initializing the action (e.g. verifying if the goal is valid and should be accepted.)

This message might actually be removed based on the usage of it.

File: penelope_aerospace_pl_msgs/action/DvmInspect.action [hunk 4]
```diff
@@ -57,6 +57,15 @@ float32[] time_to_completion
 # Refer to penelope_aerospace_pl_msgs/msg/DvmState for the definition
 DvmState[] measurement_state
 
+# Triggers state
+TriggersState triggers_state
+
+# speed between movements
```
Comment (dave992) [inline penelope_aerospace_pl_msgs/action/DvmInspect.action:63]: No live velocity feedback is coming from the Gantrry, so this suggestion is just to clarify this is a requested value and not the actual live value. 
```suggestion
# requested speed for the current movement
```

File: penelope_aerospace_pl_msgs/msg/TriggersState.msg (hunk 1 above)
Comment (djakac) [reply penelope_aerospace_pl_msgs/msg/TriggersState.msg:9]: That's ok for me

File: penelope_aerospace_pl_msgs/action/DvmInspect.action (hunk 4 above)
Comment (djakac) [reply penelope_aerospace_pl_msgs/action/DvmInspect.action:63]: ok

File: penelope_aerospace_pl_msgs/msg/ModuleState.msg (hunk 3 above)
Comment (djakac) [reply penelope_aerospace_pl_msgs/msg/ModuleState.msg:19]: I updated the code in the penelope_TNO_module. Now it's using the EXECUTING. could you fix the ModuleState.msg?

File: penelope_aerospace_pl_msgs/msg/TriggersState.msg (hunk 1 above)
Comment (djakac) [reply penelope_aerospace_pl_msgs/msg/TriggersState.msg:9]: I fixed the code in the penelope_dvm_module to use the TRIGGERS_ACTIVE and NO_TRIGGERS

File: penelope_aerospace_pl_msgs/msg/ResultCodes.msg (hunk 2 above)
Comment (djakac) [reply penelope_aerospace_pl_msgs/msg/ResultCodes.msg:16]: I updated the code in the penelope_TNO_module. Now it's using the RC_DVM_FAILED_BY_USER.

**Comment by djakac** [comment]
I have no complaints. You can merge the suggestions at least in the feature/integration_p1. And in the main if no one else has any suggestions.

**Pull request merged**